packages/web/components/Dashboard/Settings/Subscription/PaymentForm.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState, useMemo } from 'react'
+import { toast } from 'react-toastify'
+import nProgress from 'nprogress'
+import { loadStripe, StripeError } from '@stripe/stripe-js'
+import { Elements, CardElement, useStripe, useElements } from '@stripe/react-stripe-js'
+import {
+  usePurchaseMembershipSubscriptionMutation,
+  useUpdateSubscriptionPaymentMethodMutation,
+  MembershipSubscriptionPeriod,
+} from '@/generated/graphql'
+import Button from '@/components/Button'
+import SubscriptionPlanSelect from './SubscriptionPlanSelect'
+import { useTranslation } from '@/config/i18n'
+import theme from '@/theme'
+
+type PaymentFormProps = {
+  onSuccess: () => void
+  isUpdatingCard?: boolean
+}
+
+const PaymentForm = ({ onSuccess, isUpdatingCard = false }: PaymentFormProps) => {
+  const { t } = useTranslation('settings')
+  const [stripeError, setStripeError] = useState<StripeError>()
+  const stripe = useStripe()
+  const elements = useElements()
+  const [selectedOption, setSelectedOption] = useState<MembershipSubscriptionPeriod>(
+    MembershipSubscriptionPeriod.Monthly,
+  )
+
+  const [purchaseMembershipSubscription, { loading }] = usePurchaseMembershipSubscriptionMutation({
+    onCompleted: () => {
+      onSuccess()
+      toast.success(t('subscription.subscribeSuccessMessage'))
+    },
+    onError: () => {
+      toast.error(t('subscription.subscribeFailureMessage'))
+    },
+  })
+
+  const [updatePaymentMethod] = useUpdateSubscriptionPaymentMethodMutation({
+    onCompleted: () => {
+      onSuccess()
+      toast.success(t('subscription.updateCardSuccess'))
+    },
+    onError: () => {
+      toast.error(t('subscription.updateCardError'))
+    },
+  })
+
+  const handleSubmitPaymentForm = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    nProgress.start()
+    // 1. Create payment method via stripe
+    //    Token comes back here if successful
+    if (elements && stripe) {
+      const card = elements.getElement(CardElement)
+
+      if (!card) {
+        // TODO: figure out actual user messages
+        throw new Error("Card element not found")
+      }
+
+      const { error, paymentMethod } = await stripe.createPaymentMethod({
+        type: 'card',
+        card,
+      })
+      if (error) {
+        setStripeError(error)
+        nProgress.done()
+        return
+      }
+  
+      if (isUpdatingCard) {
```

Comment:
I get wanting to re-use this form for card updates, but there's a few other things I think we should do if we take that approach like change the CTA ("subscribe" doesn't really fit when you're just changing your card) and not rendering the billing period select. I would personally opt for two different components, even though there's going to be some reused code, but this approach is reasonable. I think there are just some other UI things that need to be driven by `isUpdatingCard` if we take this route.

Reply:
Yep that sounds good to me! To be honest this refactor/approach took place in about 3 minutes trying to quickly get the work across before that evening trip to Ikea 😅 So was just my best quick effort at the time. I'm all for keeping these components as clean & simple as possible so will happily refactor things following your suggestion

Reply:
@Lanny refactor is done with [this commit](https://github.com/Journaly/journaly/pull/502/commits/8b1ca9a5abcb3c023d533617aab12885770b473e)

Reply:
[One more commit](https://github.com/Journaly/journaly/pull/502/commits/4a08515908fb973bd2593346cc408be397a2f759) to finish implementing things. Let me know if this level of separation is what you had in mind 😊